cmd/github-ldap-user-group-creator/main.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/logrusutil"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	userv1 "github.com/openshift/api/user/v1"
+
+	"github.com/openshift/ci-tools/pkg/api"
+)
+
+type options struct {
+	kubernetesOptions flagutil.KubernetesOptions
+
+	mappingFile string
+	logLevel    string
+	dryRun      bool
+}
+
+func parseOptions() *options {
+	opts := &options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	opts.kubernetesOptions.AddFlags(fs)
+	fs.StringVar(&opts.logLevel, "log-level", "info", fmt.Sprintf("Log level is one of %v.", logrus.AllLevels))
+	fs.BoolVar(&opts.dryRun, "dry-run", true, "Whether to run the controller-manager with dry-run")
+	fs.StringVar(&opts.mappingFile, "mapping-file", "", "File to the mapping results of m(github_login)=kerberos_id.")
```

Comment:
Why this tool can't generate the mapping itself?

Reply:
The tool which generates the mapping has to run inside RH network (the info to generate the mapping is RH LDAP server, which is inside the RH network).
The mapping itself is stored in our cluster though.
We want it as simple as possible: Just generate the mapping, no other logic is there.

Reply:
Does the tool that generates the mapping, that runs inside RH network, can reach the clusters?

Reply:
Yes as long as the machine hosting the tool has Internet.